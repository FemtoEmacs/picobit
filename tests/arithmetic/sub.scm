(define-macro (test-cond condition name) ;; TODO watch out for name capture
  `(let ((DUMMY-x ,condition))
     (display ,name)
     (if DUMMY-x
       (display " : PASSED\n")
       (begin (display " : FAILED, got : ")
	      (display DUMMY-x)
	      (display "\n")))))

(test-cond (= (- 2 2) 0) "2 - 2 = 0")
(test-cond (= (- 5 3) 2) "5 - 3 = 0")
(test-cond (= (- 5 7) -2) "5 - 7 = -2")
(test-cond (= (- 4 0) 4) "4 - 0 = 4")
(test-cond (= (- 0 4) -4) "0 - 4 = -4")
(test-cond (= (- 2 -2) 4) "2 - -2 = 4")
(test-cond (= (- -2 -5) 3) "-2 - -5 = 3")
(test-cond (= (- -5 -2) -3) "-5 - -2 = -3")
