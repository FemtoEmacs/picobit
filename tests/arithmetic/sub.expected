:0080000080
:10800000FBD78C028189400081444000814E4000B2
:108010000003FFFE0003FFFB815640008159400032
:1080200081614000816440008165400081794000A9
:1080300081804000818D40008137400081474000B1
:10804000814940000003FFFC0003FFFD811B40004D
:10805000811940008024015380310154807800024E
:108060008038015580360157802401588024015AF8
:108070008041015B8066015C8031015D807D015E35
:108080008031015F8024016080240162803700021A
:1080900080240163803401578038000280500166DB
:1080A0008024016780570168802401698078016B12
:1080B0008041016A8070016B8041016C8024016DF8
:1080C0008031016E8024016F802401708024017151
:1080D0008027017280310173803201748041017503
:1080E0008045017680240177804501788069017A96
:1080F000803B017B8066017C8038017D8073017E3E
:108100008039017F80240181803901828040018390
:108110008039018480400185803401868048018750
:1081200080240188804801898024018A8024018B71
:108130008031018C803101298031018E8024018FB2
:108140008041013580360136802401388024013990
:108150008024013A804A013B8039013C8054013D32
:1081600080240126806E013E8024013F8071014001
:1081700080240141806B0142803101438034000240
:108180008031014580270146802701488049014A06
:108190008037014B8049014C8031014D8024014FD3
:1081A0008042000280240150803E017480240151ED
:1081B000803101528031012680310127806A00021E
:1081C000802401288024000280240129804D012A76
:1081D0008031012B8057012C8036012D8067012EC4
:1081E0008024012F80730130802401318078013296
:1081F0008024013380360134803E011E8024011F1B
:1082000080730120802401218077012280240123B2
:108210008030012480410125800E00028031011C44
:10822000803E011D803600028027011A8036011829
:108230008036014B02B4045F5002B4045951060663
:10824000C204C702B4043B61EE0907C206C702B408
:10825000041D61EE090BC2A107C702B403FF61EE62
:108260000804C208C702B403E161EE0408C2A11405
:10827000C702B403C361EE06A107C208C702B40374
:10828000A561EEA107A108C207C702B4038761EE8A
:10829000A108A107C2A115C702B4029F61EED801CF
:1082A000A10BB589EE20B30377A105B6800120DAD2
:1082B000B8954121DBB6800221CC9A21CE2161EE16
:1082C0002022CFB67200EF20B6800120DA9EA026D1
:1082D00007E620B558EEA02607E600EF20C0B8BF9D
:1082E00020B582B648012004C8B8AEA03121B5A49B
:1082F00002B583CDDCEF02A034210EC5C122CD2111
:1083000002CDB403067101210EC89220EF210E4068
:108310006221B662010421C2EF2002B559B755208F
:10832000CCB8B8A02C07E620CEB51FEE20CFB68083
:108330000120D296A02D07E600EF20CC9DA02407B7
:10834000E620CEB505EE20CFB666A109B002ADEEAF
:1083500020B002CAEEA02D07E600EF20D995A10DAE
:10836000B102AD20F09C2095A110B102ADA111B1D8
:1083700002ADA10CB102ADA106B002ADEE20B0027B
:10838000ADEEA104B102AD01A117B002ADEE209592
:10839000A105B102ADA106B002ADEE20B002ADEE76
:1083A000A104B102AD01A10EB002ADEE2095A10570
:1083B000B102ADA106B002ADEE20B002ADEEA10457
:1083C000B102AD01A113B002ADEE2095A105B1023D
:1083D000ADA106B002ADEE20B002ADEEA104B10237
:1083E000AD01A116B002ADEE2095A105B102ADA17F
:1083F00006B002ADEE20B002ADEEA104B102AD01B7
:10840000A112B002ADEE2095A105B102ADA106B05A
:1084100002ADEE20B002ADEEA104B102AD01A10AA1
:10842000B002ADEE2095A105B102ADA106B002AD3E
:10843000EE20B002ADEEA104B102AD01A10FB00279
:10844000ADEE2095A105B102ADA106B002ADEE20C2
:10845000B002ADEEA104B102AD012007E600EF02CB
:108460002004C8212302CDCDB4046C71012204C8BC
:1084700022242302CDCDCDB4047B710123B8DC21AD
:10848000B0031423222602CDCDCDB4048E71012277
:10849000B8C621B0031421242602CDCDCDB404A149
:1084A00071012022242602CDCDCDCDB404AF7000C1
:1084B0002121C4232502CDCDB404BC710122B8937F
:1084C000219220EF2298019420B1031420EF21B8CB
:0E84D0007BB77500B86EB76A21B73B21B725A0
:00000001FF
